plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/DatasetTree/Folder.tsx

```diff
@@ -26,13 +27,13 @@ const Folder: React.FC<Props> = ({
   const [isOpen, open] = useState(false);
 
   useEffect(() => {
-    if (expandAll || expandedFolders.find(item => (item.id ? item.id === id : item.name === name)))
+    if (expandAll || expandedFolders?.find(item => (item.id ? item.id === id : item.name === name)))
```

Comment:
Is there any chance the item doesn't have an ID? Are you not adding an ID to all catalog groups/items when processing the catalog (I think in the api.ts file??)??
If I am correct, do we really need item.name? We might be able to simplify expandedFolders to be just an array of IDs?

Reply:
In order to add functionality to the about data button, I use the `item.path` to expand the parent folders of the selected dataset. 
`item.path` is an array of folder names which is `item.name` and it doesn't have the item's id. that is why I expand using the `item.id` and if there is no id then I use the `item.name`.
I hope that makes sense.